rust/src/adapter/honeycomb.rs

```diff
@@ -84,8 +89,18 @@ impl HoneycombAdapter {
                 }
             }
             Event::Alloc(a) => {
-                let mut span = Span::new(trace_id, parent_id, "allocation".to_string(), a.ts, a.ts);
-                span.add_attribute_i64("amount".to_string(), a.amount.into());
+                let mut span = OtelFormatter::new_span(
+                    trace_id,
+                    parent_id,
+                    "allocation".to_string(),
+                    a.ts,
+                    a.ts,
+                );
+                OtelFormatter::add_attribute_i64_to_span(
+                    &mut span,
+                    "amount".to_string(),
+                    a.amount.into(),
+                );
```

Comment:
This is my bad (fixed in Go and JS, but i guess i forgot in Rust), but we need to fix this in all the adapters.. memory allocation events should pop the last function from the stack and add the allocation as an attribute to that, using the number of pages (the `a.amount`) as the value. Then, push that function back onto the stack and carry on